src/Nelmio/Alice/Instances/Populator/Methods/ArrayAdd.php

```diff
@@ -84,4 +85,37 @@ private function findAdderMethod($object, $property)
             return $method;
         }
     }
+
+    /**
+     * @param string $plural
+     *
+     * @return string
+     */
+    private function singularize($plural)
+    {
+        if (null === $this->singularizeClass || null === $this->singularizeMethod) {
+            // default to false so we only try this once
+            $this->singularizeClass = false;
```

Comment:
Because this way it only has to be done once. Reflection is a costly operation, in terms of performance. I tend to always minimise the amount of reflection being done. This is assuming that an instance of this class is being used for multiple calls. If not then a static class property would be better, I think.